Public/Src/Engine/Scheduler/PipExecutor.cs

```diff
@@ -1404,6 +1406,25 @@ private static void MakeSharedOpaqueOutputIfNeeded(IPipExecutionEnvironment envi
 
                                 ++retryCount;
 
+                                if (result.PipProperties != null)

```

Comment:
>if (result.PipProperties != null) [](start = 32, length = 33)

What if in the current run `result.PipProperties` turns out to be `null`? In that case, you may lose the pip properties from previous runs.

Reply:
Good catch, updated.